### PR TITLE
Remove (base64) 'REDACTED' passwords from user records.

### PR DESCRIPTION
**Pull request by rsgowman**
These values *look* like passwords hashes, but aren't, leading to
potential confusion.

Additionally, added docs to CONTRIBUTING.md detailing how to add the
permission that causes password hashes to be properly returned as well
as adjusting the test failure message should the developer not add that
permission.

b/141189502

### CODE REVIEW

File: src/auth/user-record.ts [hunk 1]
```diff
@@ -173,7 +178,16 @@ export class UserRecord {
       providerData.push(new UserInfo(entry));
     }
     utils.addReadonlyGetter(this, 'providerData', providerData);
-    utils.addReadonlyGetter(this, 'passwordHash', response.passwordHash);
+
+    // If the password hash is redacted (probably due to missing permissions)
+    // then clear it out, similar to how the salt is returned. (Otherwise, it
+    // *looks* like a b64-encoded hash is present, which is confusing.)
+    if (response.passwordHash === B64_REDACTED) {
+      utils.addReadonlyGetter(this, 'passwordHash', undefined);
```
Comment (bojeil-google) [inline src/auth/user-record.ts]: Please add a unit test for this. When a user record is initialized with a server response containing a redacted password, confirm it is cleared.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -153,13 +153,19 @@ Then set up your Firebase/GCP project as follows:
...
-   "Email/Password" sign-in method.
+   "Email/Password" sign-in method, including the Email link (passwordless
+   sign-in) option.
 3. Enable the IAM API: Go to the
    [Google Cloud Platform Console](https://console.cloud.google.com) and make
    sure your Firebase/GCP project is selected. Select "APIs & Services >
    Dashboard" from the main menu, and click the "ENABLE APIS AND SERVICES"
    button. Search for and enable the "Identity and Access Management (IAM)
    API".
+4. Grant your service account the 'Firebase Authentication Admin' role:
```
Comment (hiranya911) [inline CONTRIBUTING.md]: Can we also add a note here indicating why this permission is needed? 

"This is required to ensure that exported user records contain the password hashes of the user accounts."

Comment (rsgowman) [reply CONTRIBUTING.md]: Done.

File: src/auth/user-record.ts (hunk 1 above)
Comment (rsgowman) [reply src/auth/user-record.ts]: Done.

Comment (bojeil-google) [reply src/auth/user-record.ts]: Thanks for adding to auth.spec.ts. Can you also add to user-record.spec.ts, the test associated with this class?

Comment (rsgowman) [reply src/auth/user-record.ts]: Done. I've removed the test from auth.spec.ts, as I think with this change, it's become mostly redundant. Though we can add it back in if you'd prefer to keep it.

File: test/unit/auth/user-record.spec.ts [hunk 3]
```diff
@@ -653,6 +653,15 @@ describe('UserRecord', () => {
     });
   });
 
+  describe('clears REDACTED passwordHash', () => {
```
Comment (bojeil-google) [inline test/unit/auth/user-record.spec.ts]: This seems like a test of `passwordHash` getter.
Can you move it next to `it('should return expected undefined passwordHash', () => {`
You can define something like: `it('should clear REDACTED passwordHash', () => {`

Comment (rsgowman) [reply test/unit/auth/user-record.spec.ts]: Done

**Pull request merged**